shell/shared/grain.js

```diff
@@ -386,6 +389,57 @@ Meteor.methods({
       return outerResult;
     }
   },
+  requestAccess: function (origin, grainId, identityId) {
+    check(origin, String);
+    check(grainId, String);
+    check(identityId, String);
+    if (!this.isSimulation) {
+      const grain = Grains.findOne(grainId);
+      if (!grain) {
+        throw new Meteor.Error(404, "No such grain");
+      }
+      const grainOwner = globalDb.getUser(grain.userId);
+      const email = _.findWhere(SandstormDb.getUserEmails(grainOwner), {primary: true});
+      if (!email) {
+        throw new Meteor.Error("no email", "Grain owner has no email address.");
+      }
+      const emailAddress = email.email;
+
+      const identity = globalDb.getIdentity(identityId);
+      globalDb.addContact(grainOwner._id, identityId);
+
+      const fromEmail = "Sandstorm server <no-reply@" + HOSTNAME + ">";
```

Comment:
If you want this to come from the server, use `db.getServerTitle()` and `db.getReturnAddress()`.

But note that you could actually make the "from" be the sender's address as long as the envelope sender is the Sandstorm server. See e.g. how we send mail from apps.